app/javascript/src/Users/components/FeatureAccessInput.jsx

```diff
@@ -0,0 +1,96 @@
+// @flow
+
+import * as React from 'react'
+
+import { getFeatureName, isServicePermissionsGranted } from 'Users/utils'
+
+import type { Feature, AdminSections } from 'Users/types'
+
+type Props = {
+  features: Feature[],
+  checkedAdminSections: AdminSections[],
+  showServices?: boolean,
+  onChange: AdminSections => void
+}
+
+const FeatureAccessInput = ({ features, checkedAdminSections, showServices = false, onChange }: Props) => {
+  const olClass = `FeatureAccessList ${showServices ? '' : 'FeatureAccessList--noServicePermissionsGranted'}`
+  const allServicesChecked = !checkedAdminSections.includes('services')
+
+  return (
+    <fieldset>
+      <legend className='label'><label>This user can access</label></legend>
+      <ol className={olClass}>
+        <input type='hidden' name='user[member_permission_ids][]' />
+        {features.map(feature => (
+          <FeatureAccess key={feature} value={feature} checked={checkedAdminSections.includes(feature)} onChange={onChange}>
+            {getFeatureName(feature)}
+          </FeatureAccess>
+        ))}
+        <ServiceFeatureAccess checked={allServicesChecked} onChange={onChange}>
+          All current and future APIs
+        </ServiceFeatureAccess>
+      </ol>
+    </fieldset>
+  )
+}
+
+const FeatureAccess = ({ value, checked = false, onChange, children }: {
+  value: Feature,
+  checked?: boolean,
+  onChange: Feature => void,
+  children?: React.Node
```

Comment:
I'd rather request here an argument `featureName` of type `string` since this is what is really needed in this component, right?

Reply:
Right, having children here does not make sense at all.